dm/pkg/checker/table_structure.go

```diff
@@ -296,88 +361,183 @@ func (c *ShardingTablesChecker) Check(ctx context.Context) *Result {
+						reMu.Unlock()
+						err = err2
+						return
+					}
+				}
 
-		for schema, tables := range schemas {
-			for _, table := range tables {
-				statement, err := dbutil.GetCreateTableSQL(ctx, db, schema, table)
-				if err != nil {
+				statement, err2 := dbutil.GetCreateTableSQL(ctx, c.dbs[instance], table.Schema, table.Name)
+				if err2 != nil {
 					// continue if table was deleted when checking
-					if isMySQLError(err, mysql.ErrNoSuchTable) {
+					if isMySQLError(err2, mysql.ErrNoSuchTable) {
 						continue
 					}
-					markCheckError(r, err)
-					r.Extra = fmt.Sprintf("instance %s on sharding %s", instance, c.name)
-					return r
+					err = err2
+					return
 				}
 
-				info, err := dbutil.GetTableInfoBySQL(statement, parser2)
-				if err != nil {
-					markCheckError(r, err)
-					r.Extra = fmt.Sprintf("instance %s on sharding %s", instance, c.name)
-					return r
+				info, err2 := dbutil.GetTableInfoBySQL(statement, p)
+				if err2 != nil {
+					err = errors.Annotatef(err2, "statement: %s", statement)
+					return
 				}
-				stmt, err := parser2.ParseOneStmt(statement, "", "")
-				if err != nil {
-					markCheckError(r, errors.Annotatef(err, "statement %s", statement))
-					r.Extra = fmt.Sprintf("instance %s on sharding %s", instance, c.name)
-					return r
+				stmt, err2 := p.ParseOneStmt(statement, "", "")
+				if err2 != nil {
+					err = errors.Annotatef(err2, "statement: %s", statement)
+					return
 				}
 
 				ctStmt, ok := stmt.(*ast.CreateTableStmt)
 				if !ok {
-					markCheckError(r, errors.Errorf("Expect CreateTableStmt but got %T", stmt))
-					r.Extra = fmt.Sprintf("instance %s on sharding %s", instance, c.name)
-					return r
+					err = errors.Errorf("Expect CreateTableStmt but got %T", stmt)
+					return
 				}
 
 				if c.checkAutoIncrementPrimaryKey {
-					passed := c.checkAutoIncrementKey(instance, schema, table, ctStmt, info, r)
+					passed := c.checkAutoIncrementKey(instance, table.Schema, table.Name, ctStmt, info, r)
 					if !passed {
-						return r
+						return
 					}
 				}
 
-				if stmtNode == nil {
-					stmtNode = ctStmt
-					firstTable = dbutil.TableName(schema, table)
-					firstInstance = instance
-					continue
-				}
-
-				checkErr := c.checkConsistency(stmtNode, ctStmt, firstTable, dbutil.TableName(schema, table), firstInstance, instance)
+				checkErr := c.checkConsistency(stmtNode, ctStmt, firstTable.String(), table.String(), firstInstance, instance)
 				if checkErr != nil {
+					reMu.Lock()
 					r.State = StateFailure
 					r.Errors = append(r.Errors, checkErr)
-					r.Extra = fmt.Sprintf("error on sharding %s", c.name)
+					r.Extra = fmt.Sprintf("error on sharding %s", c.targetTableID)
 					r.Instruction = "please set same table structure for sharding tables"
-					return r
+					reMu.Unlock()
+					return
 				}
 			}
 		}
 	}
 
+	concurrency := maxThreadNum
+	for instance := range c.tableMap {
```

Comment:
```suggestion
	for sourceID := range c.tableMap {
```
and adjust rest logic